x/ibc/02-client/keeper/keeper.go

```diff
@@ -143,14 +150,25 @@ func (k Keeper) initialize(
 	ctx sdk.Context, clientID string, clientType exported.ClientType,
 	consensusState exported.ConsensusState,
 ) (exported.ClientState, error) {
-	var clientState exported.ClientState
 	height := uint64(ctx.BlockHeight())
 
+	var (
+		clientState exported.ClientState
+		err         error
+	)
 	switch clientType {
 	case exported.Tendermint:
-		clientState = tendermint.NewClientState(clientID, height)
+		unbondingPeriod := k.stakingKeeper.UnbondingTime(ctx)
+		trustingPeriod := time.Duration(defaultTrustingPeriodConstant.Numerator) * unbondingPeriod / time.Duration(defaultTrustingPeriodConstant.Denominator)
+		clientState, err = tendermint.Initialize(
+			clientID, consensusState, trustingPeriod, unbondingPeriod, height,
```

Comment:
The `trustingPeriod` and `unbondingPeriod` should be passed in as part of the `CreateClient` msg, rather than being taken from the current chains stakingKeeper or default value.

Only if these values were `0` in the msg, should they be replaced with the default constants defined above